core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -156,47 +163,102 @@ object ConsumerGroupCommand {
       val props = if (opts.options.has(opts.commandConfigOpt)) Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt)) else new Properties()
       val channelSocketTimeoutMs = props.getProperty("channelSocketTimeoutMs", "600").toInt
       val channelRetryBackoffMs = props.getProperty("channelRetryBackoffMsOpt", "300").toInt
+      if (!zkUtils.getConsumerGroups().contains(group)) {
+        println("The consumer group provided does not exist")
+        return
+      }
       val topics = zkUtils.getTopicsByConsumerGroup(group)
       if (topics.isEmpty)
         println("No topic available for consumer group provided")
       printDescribeHeader()
-      topics.foreach(topic => describeTopic(group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
-    }
 
-    private def describeTopic(group: String,
-                              topic: String,
-                              channelSocketTimeoutMs: Int,
-                              channelRetryBackoffMs: Int) {
-      val topicPartitions = getTopicPartitions(topic)
-      val groupDirs = new ZKGroupTopicDirs(group, topic)
-      val ownerByTopicPartition = topicPartitions.flatMap { topicPartition =>
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      val topicPartitions = getTopicsPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of ephemeral owner -> owner
+      val ownerByEOwner = scala.collection.mutable.Map[Long, String]()
+      // mapping of topic partition -> ephemeral owner
+      val eOwnerByTopicPartition = scala.collection.mutable.Map[TopicAndPartition, Long]()
+      // mapping of topic partition -> owner
+      val ownerByTopicPartition = scala.collection.mutable.Map[TopicAndPartition, String]()
```

Comment:
That's fair. I'll update the code to return the output in the format you suggested. I'm thinking of always returning an empty string for client id and ip for ZK-based consumer (as you suggested), and returning "none" for member id when no active consumer exists for the group.